modules/azure-public-dns/init.tf

```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 0.12.0"
+  required_providers {
+    azurerm = ">= 1.7.0"
+  }
+}
+
+provider "azurerm" {
+  version = "=1.34.0"
```

Comment:
this shouldn't be in a module, providers should be declared at the top level

Reply:
Thanks for the spot Tim. Updated.